ex2/ex2.c

```diff
@@ -5,10 +5,47 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
-
+  #include <sys/types.h>
+  #include <sys/wait.h>
+int printStr (FILE *x){
+    while(1) {
+      int c = fgetc(x);
+      if( feof(x) ) { 
+         break ;
+      }
+      printf("%c", c);
+   }
+      printf("\n");
+    return 0;
+}
 int main(void)
 {
     // Your code here 
-    
+    FILE *test = fopen("text.txt","a+");
+    printf("Intial read:");
+    printStr(test);
+    fclose(test);
+
+    int child = fork();
+    if (child == 0){
+        FILE *test = fopen("text.txt","a+");
+        fprintf(test, "%s %s %s \n", "Child", "was", "here", 2012);
+
```

Comment:
When I compiled and ran this I got a warning saying that there were too many arguments in lines 32 and 42 for the format. Does it work correctly for you?

Reply:
Yeah it does, I suspect the issue is the 2012 and I've removed that. (Not 100% sure where it came from)